control-plane/stor-port/src/types/v0/store/volume.rs

```diff
@@ -591,3 +591,42 @@ impl From<VolumeSpec> for models::VolumeSpec {
         )
     }
 }
+
+/// In memory representation of volume group.
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct VolumeGroupSpec {
+    /// Name of the volume group.
+    volume_group_name: String,
+    /// List of ids of volumes that belong to the volume group.
+    volumes: Vec<VolumeId>,
+    /// The operation sequence resource is in.
+    sequencer: OperationSequence,
+}
+
+#[allow(dead_code)] // todo: remove this after other implementation is in place.
+impl VolumeGroupSpec {
+    /// Create a new VolumeGroupSpec.
+    pub fn new(volume_group_name: String, volumes: Vec<VolumeId>) -> Self {
+        Self {
+            volume_group_name: volume_group_name.clone(),
+            volumes,
+            sequencer: OperationSequence::new(volume_group_name),
+        }
+    }
+    /// Name of the volume group.
+    pub fn name(&self) -> &String {
+        &self.volume_group_name
+    }
+    /// Add a new volume to the list of volumes.
+    pub fn append_volume(&mut self, id: VolumeId) {
+        self.volumes.push(id)
+    }
+    /// Add a new volume to the list of volumes.
+    pub fn remove_volume(&mut self, id: VolumeId) {
+        self.volumes.retain(|vol| vol.clone() != id)
```

Comment:
why clone?

Reply:
`self.volumes.retain(|vol| vol != &id)` How about this?